network/round-robin.go

```diff
@@ -0,0 +1,20 @@
+package network
+
+import "sync/atomic"
+
+type RoundRobin struct {
+	proxies []IProxy
+	next    atomic.Uint32
+}
+
+func NewRoundRobin(server *Server) *RoundRobin {
+	return &RoundRobin{proxies: server.Proxies}
+}
+
+func (r *RoundRobin) GetNextProxy() IProxy {
+	proxiesLen := uint32(len(r.proxies))
+
+	nextIndex := r.next.Add(1)
+
+	return r.proxies[nextIndex%proxiesLen]
+}
```

Comment:
Wasn't it better to have a circular linked-list?

Reply:
Thanks for your review and suggestion! I saw your PR, and it's a really elegant solution. However, I went with an array index using sync/atomic because our proxy count is fixed. Indexing offers better performance and memory efficiency due to better cache locality and direct indexing, avoiding the pointer overhead of linked lists. Plus, sync/atomic gives us efficient thread-safe access, if we scale with more Goroutines later on.
those are my reasons for picking sync/atomic.